What's new indicator

joeyg opened the pull request:
Fixes issue #350. 

Screencast: http://recordit.co/qLwFllwClv

bkmunar commented:
amazing! I'll get around to reviewing today! and let's get this merged in to the next release :)

joeyg commented:
@bkmunar After taking a look at this I'm not crazy about the way it calls the AppDelegate from other parts of the app. Is there a pattern we have been using elsewhere?

bkmunar commented:
That's a good question; there are certain util files designed for functions, like OpenUtils, AppInfo, so maybe see if you can use one of those to store these functions? @boek pinging you for input as well

boek commented:
Hey @joeyg thanks for getting ahead of this 👍. I agree with you regarding the AppDelegate. I would love to start moving towards using dependency injection and protocols for things like this. Unfortunately, there is no existing patterns in the app. Feel free to take a stab at it, otherwise I will be adding something this week for the work I'll be doing on #343

joeyg commented:
@boek Thanks for the feedback. I'll take a stab at moving it to a protocol and using dependency injection. Is there an asset for the blue notification dot or should that be done programmatically.

boek commented:
Hey @joeyg, I don't think this is final but it should work for now :)

[Preferences Updated.pdf](https://github.com/mozilla-mobile/focus-ios/files/1369711/Preferences.Updated.pdf)

lime124 commented:
Looking good so far. It looks like we're missing a loading indicator from our other screens that load web content too (e.g. About -> Help). Let me get back to you about that.

And yes, the icon @boek sent you isn't final :)

Thanks for helping out on this issue!

joeyg commented:
Updated the PR to remove some code from AppDelegate and add in the Settings Indicator icon. 

New Screencast: http://recordit.co/8AtvNKdC0l

bkmunar commented:
It's looking really great!! The only thing I think left is, how do we know when to show What's New Again after the website has been updated and such? @boek @Sdaswani do we know how often we will be updating the website? Maybe keep showing What's New Page on each version update like Firefox does?

Sdaswani commented:
@bkmunar i expect the answer to your question can be found in the android code base, which starts with this commit: https://github.com/mozilla-mobile/focus-android/commit/fa15fc79cd07645736dc29e3e3682acf231c0206 , and maybe you can find something specified in the android ticket: https://github.com/mozilla-mobile/focus-android/issues/1201 You can also ping @pocmo for details if they can't be found.

Sdaswani commented:
But I also agree @joeyg - the feature is looking great, thanks for your contribution. You should take a look at the Android implementation as I think it differs a bit cc @antlam @lime124 http://recordit.co/8AtvNKdC0l

lime124 commented:
It looks like we just need to get you the final icon. And then figure out a loading indicator. But we might be better to open up a new issue for the loading indicator and put the issue in the parking lot since it affects more than just this view? @Sdaswani what do you think?

Sdaswani commented:
Sounds go to me @lime124, I think 'What's New' won't be in the parking lot for long anyhow.

joeyg commented:
Here is the documentation in the Android app about when it shows the WhatsNew indicator: https://github.com/mozilla-mobile/focus-android/blob/9210c6a339a1c4ec9733eb1e2becbb1daa546a11/app/src/main/java/org/mozilla/focus/whatsnew/WhatsNew.kt#L13

And here is the code for creating the URL: https://github.com/mozilla-mobile/focus-android/blob/9210c6a339a1c4ec9733eb1e2becbb1daa546a11/app/src/main/java/org/mozilla/focus/utils/SupportUtils.java#L22

lime124 commented:
made an issue for the webpage loading indicator - #575

joeyg commented:
@Sdaswani are there specific changes that need to be made to match Android’s implantation?

@boek what do you think of the updated PR in terms of code quality? Are there other changes that could tighten it up?

Sdaswani commented:
Sorry for the delay @joeyg. 

Here are next steps: @bkmunar and @boek to CR.

@lime124 will provide a page for you to point to soon.

@antlam to provide a final icon soon.

Sdaswani commented:
_and_ @BrianNJones to provide copy for the what's new page.

joeyg commented:
Thanks @Sdaswani

antlam commented:
@boek @Sdaswani 

[Preferences Updated.pdf](https://github.com/mozilla-mobile/focus-ios/files/1396164/Preferences.Updated.pdf)

PDF asset!

boek commented on Blockzilla/AppDelegate.swift:
```diff
@@ -111,7 +114,25 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+            switch counter {
+                case 0:
+                    // No counter set
+                    UserDefaults.standard.set(1, forKey: AppDelegate.prefWhatsNewCounter)
```
Do we need this case if the default will give you `0 + 1`?

joeyg replied:
good call!

boek commented on Blockzilla/BrowserViewController.swift:
```diff
@@ -50,12 +50,12 @@ class BrowserViewController: UIViewController {
     private var shouldEnsureBrowsingMode = false
     private var initialUrl: URL?
-
+    
```
nit: extra whitespace in this file

boek commented:
Hey @joeyg, looking good! Just some whitespace and a comment on the switch statement :)

joeyg commented:
@antlam is that an updated file? I'm not seeing any differences in the images.

joeyg commented:
Updated PR to address @boek pr feedback and merge conflict.

Sdaswani commented:
Thanks @boek - @antlam needs to confirm whether the 'Preference Updated.pdf' is the latest and @lime124 needs to provide the 'What's New' URL.

Sdaswani commented:
@joeyg I spoke to @antlam and that last pdf is the final image.

Sdaswani commented:
@joeyg the final url to use should be available tomorrow - really huge effort that we really appreciate!!!!

joeyg commented:
@Sdaswani Thanks! Regarding the icon, what I meant is that I don't think it changed. 

I downloaded the one from yesterday (https://github.com/mozilla-mobile/focus-ios/files/1396164/Preferences.Updated.pdf) and the one that was from 10 days ago (https://github.com/mozilla-mobile/focus-ios/files/1369711/Preferences.Updated.pdf) and I got the same md5 hash. 

Just want to make sure I have the correct asset in the PR.

Sdaswani commented:
Yes, that is the correct asset @joeyg .

joeyg commented:
@Sdaswani thanks! I’ll try and update the PR with that URL this evening.

joeyg commented:
@Sdaswani updated support URL

Sdaswani commented:
Awesome, pretty straightforward patch set. Can we get a final record.it video @joeyg so we can get approval from Product & UX?

bkmunar commented:
@bbinto @antlam @lime124 ^^^

joeyg commented:
Updated PR to not show the WhatsNew indicator on the first run after Onboarding.
